sdk/maps/Azure.ResourceManager.Maps/tests/ScenarioTests/MapsAccountTests.cs

```diff
@@ -0,0 +1,208 @@
+            Assert.AreEqual(accounts.Count, 0);
+
+            // Create accounts
+            var accountName1 = await CreateDefaultMapsAccount(MapCollection, ResourceGroup.Data.Name);
+            var accountName2 = await CreateDefaultMapsAccount(MapCollection, ResourceGroup.Data.Name);
+
+            accounts = await MapCollection.GetAllAsync().ToEnumerableAsync();
+            Assert.AreEqual(2, accounts.Count);
+
+            VerifyAccountProperties(accounts.First().Data, true, Name.S0);
+            VerifyAccountProperties(accounts.Skip(1).First().Data, true, Name.S0);
+        }
+
+        [Test]
+        public async Task MapsAccountListBySubscriptionTest()
+        {
+            // Create account
+            await SetCollection();
+            var accountName1 = await CreateDefaultMapsAccount(MapCollection, ResourceGroup.Data.Name);
+
+            // Create different resource group and account
+            var rg2 = await CreateResourceGroupAsync();
+            var mapCollection2 = rg2.GetMapsAccounts();
+            var accountName2 = await CreateDefaultMapsAccount(mapCollection2, rg2.Data.Name);
+
+            var accounts = await DefaultSubscription.GetMapsAccountsAsync().ToEnumerableAsync();
+
+            Assert.GreaterOrEqual(accounts.Count, 2);
+
+            var account1 = accounts.First(
+                t => StringComparer.OrdinalIgnoreCase.Equals(t.Data.Name, accountName1));
+            VerifyAccountProperties(account1.Data, true, Name.S0);
+
+            var account2 = accounts.First(
+                t => StringComparer.OrdinalIgnoreCase.Equals(t.Data.Name, accountName2));
+            VerifyAccountProperties(account2.Data, true, Name.S0);
+        }
+
+        [Test]
+        public async Task MapsAccountListKeysTest()
+        {
+            await SetCollection();
+
+            // prepare account properties
+            string accountName = Recording.GenerateAssetName("maps");
+            var parameters = GetDefaultMapsAccountData();
+
+            // Create account
+            var newAccount = (await MapCollection.CreateOrUpdateAsync(WaitUntil.Completed, accountName, parameters)).Value;
+            VerifyAccountProperties(newAccount.Data, true, Name.S0);
+
+            // List keys
+            var keys = (await newAccount.GetKeysAsync()).Value;
+            Assert.NotNull(keys);
+
+            // Validate Key1
+            Assert.NotNull(keys.PrimaryKey);
+
+            // Validate Key2
+            Assert.NotNull(keys.SecondaryKey);
+        }
+
+        [Test]
+        public async Task MapsAccountRegenerateKeyTest()
+        {
+            await SetCollection();
+
+            // prepare account properties
+            string accountName = Recording.GenerateAssetName("maps");
+            var parameters = GetDefaultMapsAccountData();
+
+            // Create account
+            var newAccount = (await MapCollection.CreateOrUpdateAsync(WaitUntil.Completed, accountName, parameters)).Value;
```

Comment:
any reason we cannot use the `CreateDefaultMapsAccount` method here? We are using the same parameters as `CreateDefaultMapsAccount`

Reply:
Done